networkx/algorithms/shortest_paths/weighted.py

```diff
@@ -1935,3 +1939,200 @@ def dist_path(v):
+
+    if init_sol is not None:
+        if not _check_feasible_init_sol(G, init_sol, tmax):
+            raise nx.NetworkXError('The initial solution is not feasible.')
+
+    if temp <= 0:
+        raise nx.NetworkXError('The temperature must be greater than 0.')
+
+    best_sol = list(init_sol)
+    current_sol = list(init_sol)
+    for i in range(outer_loop):
+        for j in range(inner_loop):
+            neighbor_sol = list(_moving(G, current_sol, tmax))
+            score_of_current = _score_calculation(scores, current_sol)
+            score_of_neighbor = _score_calculation(scores, neighbor_sol)
+            df = score_of_neighbor - score_of_current
+            if df >= 0:
+                current_sol = list(neighbor_sol)
+                if  _score_calculation(scores, current_sol) > _score_calculation(scores, best_sol):
+                    best_sol = list(current_sol)
+            else:
+                p = float("{0:.2f}".format(exp(df/temp)))
+                r = float("{0:.2f}".format(random()))
+                current_sol = list(neighbor_sol) if r < p else list(current_sol)
+        temp = temp - temp * a
+    best_score = _score_calculation(scores, current_sol)
+    return best_sol, best_score, _cost_calculation(G, best_sol)
+
+def _moving(G, current_sol, tmax):
+    """
+    Method to apply a movement to the current solution in order to generate the
+    neighbor solution.
+
+    :param G: The graph of the problem. A Networkx DiGraph.
+    :param current_sol: List of nodes containing the sequence of the current
+    solution's nodes
+    :param tmax: int, The fixed amount of time.
+    :return: list, A neighbor solution
+    """
+    cs = list(current_sol)
+    current_sol_backup = list(current_sol)
+    current_sol_item = randint(1, len(current_sol) - 2)
+    helping_list = []
+    for n in G.nodes():
+        if n not in current_sol:
+            helping_list.append(n)
+    a =  randint(0, len(helping_list) - 1)
+    item = helping_list[a]
+    cs[current_sol_item], helping_list[a] = helping_list[a], cs[current_sol_item]
+    return cs if _cost_calculation(G, cs) <= tmax else current_sol_backup
+
+def _cost_calculation(G, sol):
+    """
+    Method to calculate the cost (distance) of the solution.
+
+    :param G: The graph of the problem. A Networkx DiGraph.
+    :param sol: List of nodes containing the sequence with which they must be
+    visited
+    :return: float, cost of solution (distance)
+    """
+    return sum(G.edge[sol[i]][sol[i + 1]]['weight'] for i in range(len(sol) - 1))
+
+
+def _score_calculation(scores, sol):
+    """
+    Method to calculate the sum of scores for the nodes of the solution
+
+    :scores: dictionary, The scores of each node
+    :param sol: List of nodes containing the sequence with which they must be
+    visited
+    :return: int, Total  sum of scores
+    """
+    return sum(scores.get(str(i), 0) for i in sol)
+
+
+def _graph_is_full_completed(G):
```

Comment:
This single-use function can be inlined.

Also, this fails for graphs that have self-loops, so you should deal with that case, too.